### PR TITLE
Profiles: use `name` text record instead of `me.rainbow.displayName`

### PR DESCRIPTION
**Pull request by jxom**
Fixes TEAM2-432

## What changed (plus any additional context for devs)

With the `name` record to soon be a standard in ENS in a future [ENSIP](https://docs.ens.domains/ens-improvement-proposals), we should probably use that instead of `me.rainbow.displayName`. Since we can't migrate ENS text record fields for users without it costing gas, we will still need to support the `me.rainbow.displayName` field if it is set (and a `name` is not). 

This PR migrates to set the `name` field when a user is creating/updating their profile. If a user is updating their profile, and already has a `me.rainbow.displayName` set, when the profile is updated, it will migrate this field to use `name` instead (delete the `me.rainbow.displayName` record, and then set the `name` record).
This PR also adds the ability to still view the `me.rainbow.displayName` field if a user does not yet have a `name` field set (via the `deprecatedTextRecordFields` value). 

## Screen recordings / screenshots

Here is a quick video showing how a migration works when a user updates their profile (note, the migration will occur for **any** update to records).

https://www.loom.com/share/976958d2887f4c67b1c9fd974168a34d 

If a user has both a `name` and `me.rainbow.displayName` field set, it will prefer the `name` value:

Screen Shot 2022-08-19 at 9 12 41 am

Screen Shot 2022-08-19 at 9 12 58 am

## What to test

- If a user only has `me.rainbow.displayName` set, ensure you can still view the name on:
  - the View Profile screen
  - the Edit Profile screen
  - the ENS NFT expanded state
- If a user has both a `name` and `me.rainbow.displayName` set, ensure that the `name` value is preferred on:
  - the View Profile screen
  - the Edit Profile screen
  - the ENS NFT expanded state
- If a user has a `me.rainbow.displayName` set, and goes to update **any** record on their profile, after updating ensure that the `me.rainbow.displayName` record is removed, and the `name` record is now set

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

**Review by pugson — APPROVED** [review]
amazing 🥇 👏

File: e2e/registerENSFlow.spec.js [hunk 1]
```diff
@@ -408,7 +398,7 @@ describe('Register ENS Flow', () => {
       throw new Error('Resolved address is wrong');
     if (primaryName !== RAINBOW_WALLET_NAME)
       throw new Error('Resolved primary name is wrong');
-    if (displayName) throw new Error('me.rainbow.displayName name is wrong');
+    if (name) throw new Error('name name is wrong');
```
Comment (skylarbarrera) [inline e2e/registerENSFlow.spec.js:401]: name name 🤯

File: src/helpers/ens.ts [hunk 2]
```diff
@@ -369,6 +370,12 @@ export const textRecordFields = {
   [key in ENS_RECORDS]?: TextRecordField;
 };
 
+export const deprecatedTextRecordFields = {
```
Comment (skylarbarrera) [inline src/helpers/ens.ts:373]: clean af

**Review by skylarbarrera — APPROVED** [review]
tested all scenarios, code looks crisp 🍎 


PoW: https://cloud.skylarbarrera.com/Screen-Shot-2022-08-19-10-37-34.89.png

File: e2e/registerENSFlow.spec.js [hunk 3]
```diff
@@ -408,7 +398,7 @@ describe('Register ENS Flow', () => {
       throw new Error('Resolved address is wrong');
     if (primaryName !== RAINBOW_WALLET_NAME)
       throw new Error('Resolved primary name is wrong');
-    if (displayName) throw new Error('me.rainbow.displayName name is wrong');
+    if (name) throw new Error('name is wrong');
```
Comment (estebanmino) [inline e2e/registerENSFlow.spec.js:401]: 👌

**Review by estebanmino — COMMENTED** [review]
code is looking good! nice solution to get this to work, just want to make a note that making this will cost a bit more to update the profile, talked with @jxom directly about it 🤝 

i also found this https://www.loom.com/share/cecf1699694b4929b24a7775e631634a when migrating the name to the new format is taking the old name `Esteban` as updated when I edit the name and putting `Esteban` -> `Esteba` -> `Esteban` again, you can see the review button
 probably edge case but this doesn't happen on other records

**Comment by jxom** [comment]
I can't seem to reproduce this :\

https://user-images.githubusercontent.com/7336481/186576400-805e5e5b-b5f0-41a3-8ed6-b8f1d4ccdbb2.mp4

**Comment by estebanmino** [comment]
@jxom maybe because i'm doing the migration on the video so i didn't have the new name record? seems like moxey.eth is already migrated

**Review by estebanmino — APPROVED** [review]
working perfectly! 🚀

**Comment by BrodyHughes** [comment]
I was having trouble QAing bc of the bug where collections don't show up, but I think were gonna merge then QA spot check in the release branch for 1.7.4.

**Pull request merged**